static/js/components/widgets/MenuField.tsx

```diff
@@ -0,0 +1,366 @@
+import React, { useCallback, useState } from "react"
+import Nestable, { Item } from "react-nestable"
+import * as R from "ramda"
+import BasicModal from "../BasicModal"
+import MenuItemForm, { MenuItemFormValues } from "../forms/MenuItemForm"
+import Dialog from "../Dialog"
+import { generateHashCode, isUuid4 } from "../../lib/util"
+
+interface MenuFieldProps {
+  name: string
+  value: HugoItem[]
+  onChange: (event: Event) => void
+  collections?: string[]
+}
+
+export type HugoItem = {
+  identifier: string
+  name: string
+  url?: string
+  weight: number
+  parent?: string
+}
+
+export type InternalSortableMenuItem = {
+  id: any
+  text: string
+  targetContentId: string | null
+  targetUrl: string | null
+  children?: InternalSortableMenuItem[]
+}
+
+type onChangeProps = {
+  items: Item[]
+  dragItem: Item
+  targetPath: number[]
+}
+
+type renderItemProps = {
+  collapseIcon: React.ReactNode
+  depth: number
+  handler: React.ReactNode
+  index: number
+  item: Item
+}
+
+const topLevelKey = "_"
+
+const compareHugoValues = (item1: HugoItem, item2: HugoItem): number =>
+  item1.weight < item2.weight ? -1 : 1
+
+const hugoItemToInternal = (item: HugoItem): InternalSortableMenuItem => {
+  const partialHugoItem: {
+    targetContentId: string | null
+    targetUrl: string | null
+  } = {
+    targetContentId: null,
+    targetUrl:       null
+  }
+  if (isUuid4(item.identifier)) {
```

Comment:
Is it possible to add another field `menuItemType` to the `HugoItem`? Or if that's not possible, can we change the identifier to have a prefix like `external-...` and `internal-...` so that this is more explicit?

Reply:
Not a bad idea, but that prefix would need to be accounted for in a bunch of different places (here, in the form component, in the back end). We do get a slight advantage of brevity with the existing approach. Given the time crunch I think this is something we could address later

Reply:
I'm fine leaving it but IMO it is very unintuitive and we won't have time to fix this later either

Reply:
@noisecapella Thought of a good compromise that doesn't involve a huge change: I'll prefix the external links. The absence of the prefix on internal links should be a helpful indicator.

Reply:
Thanks. Will we need to do something in the future to support internal links to other websites within ocw-studio? Not sure if there is a use case or not

Reply:
I would consider that an external link. We haven't discussed anything about linking to other sites within Studio so I don't think that needs to be part of this PR